### PR TITLE
CMake support

### PR DESCRIPTION
**Pull request by ligurio**

### CODE REVIEW

**Comment by ligurio** [comment]
@nealcardwell, could you review the PR?

**Comment by nealcardwell** [comment]
Hi,

Thanks for your contribution. Two thoughts:

(1) Can you please update the commit messages for these commits to have the format described here (including the first line format and the signed-off-by):
  https://github.com/google/packetdrill#how-to-submit-a-patch-for-packetdrill

(2) Can you please include instructions on how to download the version of cmake that your patch assumes? When I attempt to use cmake from Ubuntu 16.04.6 LTS  with "sudo apt-get install cmake" I get the following error:

host:~/packetdrill/gtests/net/packetdrill/build$ cmake ..
CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.9.2 or higher is required.  You are running version 3.5.1
-- Configuring incomplete, errors occurred!

Thanks!

**Comment by ligurio** [comment]
Neal,

Thanks for review.

> (1) Can you please update the commit messages for these commits to have the format described here (including the first line format and the signed-off-by):
> https://github.com/google/packetdrill#how-to-submit-a-patch-for-packetdrill

Patch has been updated.

> (2) Can you please include instructions on how to download the version of cmake that your patch assumes? When I attempt to use cmake from Ubuntu 16.04.6 LTS with "sudo apt-get install cmake" I get the following error:

> host:~/packetdrill/gtests/net/packetdrill/build$ cmake ..
> CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
> CMake 3.9.2 or higher is required. You are running version 3.5.1
> -- Configuring incomplete, errors occurred!

Looks like cmake 3.5.1 is more than enough to build packetdrill, so I decreased required version in CMakeLists.txt. Although you can always install newer cmake from [binary packages](https://github.com/Kitware/CMake/releases/) or build it yourself using sources - https://cmake.org/install/

**Comment by ligurio** [comment]
Ping

File: gtests/net/packetdrill/README [hunk 1]
```diff
@@ -23,6 +23,11 @@ Then build the tool:
 
 #  make
 
+To build packetdrill with CMake:
```
Comment (nealcardwell) [inline gtests/net/packetdrill/README]: Regarding the instructions about how to build packetdrill with CMake: when I followed those instructions it did not build packetdrill. I found I needed to add one more command:

   make

Did I do something wrong? Or is this "make" line just missing from the recipe? Thanks!

**Comment by nealcardwell** [comment]
Thanks!

**Pull request merged**